test/scenarios/compile/compile.js

```diff
@@ -0,0 +1,286 @@
+var MemoryLogger = require("../memorylogger");
+var CommandRunner = require("../commandrunner");
+var fs = require("fs");
+var path = require("path");
+var assert = require("assert");
+var Server = require("../server");
+var Reporter = require("../reporter");
+var sandbox = require("../sandbox");
+var log = console.log;
+
+describe("Repeated compilation of contracts with inheritance", function() {
+  var config;
+  var contracts;
+  var contractPaths;
+  var artifactPaths;
+  var initialTimes;
+  var finalTimes;
+  var output;
+  var mapping = {};
+
+  var project = path.join(__dirname, '../../sources/inheritance');
+  var names = ["Root", "Branch", "LeafA", "LeafB", "LeafC", "SameFile1", "SameFile2", "LibraryA"];
+  var logger = new MemoryLogger();
+
+  // ----------------------- Utils -----------------------------
+  function processErr(err, output){
+    if (err){
+      log(output);
+      throw new Error(err);
+    }
+  }
+
+  function waitSecond() {
+    return new Promise((resolve, reject) => setTimeout(() => resolve(), 1250));
+  }
+
+  function getContract(key) {
+    return fs.readFileSync(mapping[key].contractPath);
+  }
+
+  function getArtifact(key) {
+    return fs.readFileSync(mapping[key].artifactPath);
+  }
+
+  function getArtifactStats() {
+    const stats = {};
+    names.forEach(key => {
+      const mDate = fs.statSync(mapping[key].artifactPath).mtime.getTime();
+      stats[key] = mDate;
+    });
+    return stats;
+  }
+
+  function touchContract(key, file) {
```

Comment:
Can we get rid of `file` as an argument here? Not that it's a big deal, but might save some travis-time